### DISCUSSION TITLE
Pen+ 3d engine!

### DISCUSSION THREAD

**Discussion by David-Orangemoon** [TurboWarp/extensions#136] [Show and tell]
I have updated the 3d engine I made with pen+ to work with the new version of the extension(that has not been merged yet). To work with perspective correct texture warping
![image_2023-01-10_195825803](https://user-images.githubusercontent.com/76855369/211693441-4a7230a2-64a1-48bf-bde8-60b32894885f.png)
![image_2023-01-10_195830350](https://user-images.githubusercontent.com/76855369/211693452-725adf44-5dce-46b3-9fb7-cff500a40687.png)
![image_2023-01-10_195834288](https://user-images.githubusercontent.com/76855369/211693459-742923bc-1f0d-4d7e-b40c-6dc0ea8fd4fb.png)
[video (2).webm](https://user-images.githubusercontent.com/76855369/211693463-22eab8c0-9e38-44f3-b080-183ba2011436.webm)

**Comment by Sammyueru** [comment]
cool

**Comment by Forysium** [comment]
holy shit this is insane! Never seen something like this made with scratch/turbowarp

**Comment by PSYclownYT** [comment]
could I have the .sb3?

**Reply by David-Orangemoon** [reply]
I will see if I have it still

**Reply by zaxhs** [reply]
did you have it or no?

**Comment by snowboyz0825** [comment]
this is actually insane I barely believe it's actually in turbowarp

**Comment by David-Orangemoon** [comment]
> this is actually insane I barely believe it's actually in turbowarp

This is an older version
https://youtu.be/xk7JcgJVEQs?si=bVBHd_RUX1grChqs

**Reply by no1anator** [reply]
could you include a .sb3 download on the game jolt site as well instead of just the link?

**Comment by LegoBrainBiker** [comment]
consider adding this to [pen+ featured projects](https://pen-group.github.io/Featured/)
